components/faction/commons/faction.lua

```diff
@@ -64,6 +66,36 @@ function Faction.read(faction)
 		or Faction.aliases[faction]
 end
 
+--- Parses multiple factions from input.
+-- Returns a table of the found factions short handle/identifier.
+-- Returns empty table if no input is specified.
+---@props props {input: string, sep: string?}
+---@return table
+function Faction.readMultiFaction(props)
+	if String.isEmpty(props.input) then
+		return {}
+	end
+
+	if Faction.read(props.input) then
+		return {Faction.read(props.input)}
+	end
+
```

Comment:
I think these lines are not needed since they get covered by the code below

Reply:
the string.isempty check maybe but why would the faction.read be redundant?
the code below seps them either by given sep. or goes char for char through the string

Reply:
Ah, you're right.

Btw sc2's Faction.read right now maps 'pz' to 'p'. That will need adjusting I think.

Reply:
have catch nil too, else `:gmatch` or `mw.text.split` error